libraries/cmake/source/lzma/CMakeLists.txt

```diff
@@ -90,20 +77,11 @@ function(lzmaMain)
     "${library_root}/src/liblzma/simple/simple_coder.c"
     "${library_root}/src/liblzma/simple/simple_decoder.c"
     "${library_root}/src/liblzma/simple/simple_encoder.c"
+    "${library_root}/src/liblzma/simple/armthumb.c"
+    "${library_root}/src/liblzma/simple/arm.c"
+    "${library_root}/src/liblzma/simple/x86.c"
   )
 
-  if("${TARGET_PROCESSOR}" STREQUAL "aarch64")
-    target_sources(thirdparty_lzma PRIVATE
-      "${library_root}/src/liblzma/simple/armthumb.c"
-      "${library_root}/src/liblzma/simple/arm.c"
-    )
-
-  elseif("${TARGET_PROCESSOR}" STREQUAL "x86_64")
-    target_sources(thirdparty_lzma PRIVATE
-      "${library_root}/src/liblzma/simple/x86.c"
-    )
-  endif()
-
```

Comment:
Why the logic has been removed? Isn't this processor specific?

Reply:
Actually these decoders should not be platform specific, yet it is unlikely osquery code will have to make use of them.